src/new/wallet.js

```diff
@@ -0,0 +1,441 @@
+        allTokens.push(d.token);
+      }
+    }
+
+    for (const d of data) {
+      const dataOutput = {'address': d.address, 'value': d.value, 'tokenData': tokens.getTokenIndex(allTokens, d.token.uid)};
+      const partialData = {'inputs': [], 'outputs': [dataOutput]}
+      const result = wallet.prepareSendTokensData(partialData, d.token, true, historyTxs, allTokens);
+
+      if (!result.success) {
+        console.log('Error sending tx:', result.message);
+        return;
+      }
+
+      const dataToken = result.data;
+
+      txData['inputs'] = [...txData['inputs'], ...dataToken['inputs']];
+      txData['outputs'] = [...txData['outputs'], ...dataToken['outputs']];
+    }
+
+    txData.tokens = tokens.filterTokens(allTokens, HATHOR_TOKEN_CONFIG).map((token) => token.uid);
+
+    const updateOutputs = (val) => {
+      const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+      for (const input of txData.inputs) {
+        historyTxs[input.tx_id]['outputs'][input.index]['spent_by'] = val;
+      }
+    }
+
+    // TODO XXX Setting input as spent by, so it won't be chosen again
+    updateOutputs('2');
+
+    const promise = new Promise((resolve, reject) => {
+      transaction.sendTransaction(txData, this.pinCode).then((result) => {
+        if (result.success === false) {
+          // XXX Setting outputs as spent_by null again
+          updateOutputs(null);
+        }
+        resolve(result);
+      }, (error) => {
+        // XXX Setting outputs as spent_by null again
+        updateOutputs(null);
+        reject(error);
+      });
+    })
+
+    return promise;
+  }
+
+  /**
+   * Send tokens to only one address.
+   *
+   * @param {String} address Address to send the tokens
+   * @param {number} value Amount of tokens to be sent
+   * @param {Object} token Token object {'uid', 'name', 'symbol'}. Optional parameter if user already set on the class
+   *
+   * @return {Promise} Promise that resolves when transaction is sent
+   **/
+  sendTransaction(address, value, token) {
+    const txToken = token || this.token;
+    if (!txToken) {
+      throw Error('Token must be set in the class instance or passed as parameter in the method.')
+    }
+    const isHathorToken = txToken.uid === HATHOR_TOKEN_CONFIG.uid;
+    // XXX This allow only one token to be sent
+    // XXX This method allow only one output
+    const tokenData = (isHathorToken ? 0 : 1);
+    const data = {
+      tokens: isHathorToken ? [] : [txToken.uid],
+      inputs: [],
+      outputs: [{
+        address, value, tokenData
+      }],
+    };
+
+    const walletData = wallet.getWalletData();
+    const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+    const ret = wallet.prepareSendTokensData(data, txToken, true, historyTxs, [txToken]);
+
+    if (!ret.success) {
+      return Promise.reject(ret.message);
+    }
+
+    return transaction.sendTransaction(ret.data, this.pinCode);
+  }
```

Comment:
I think we're missing something like `updateOutputs` like we have on `sendMultiTokenTransaction`.

Reply:
Actually, I think it's easier if this function just calls `sendMultiTokenTransaction`. We'd have just one code to maintain.

Reply:
`updateOutputs` was used in `sendMultiTokenTransaction` so we can have multiple outputs of the same token without choosing the same input. In this method we update the `spent_tx` when the created tx arrives on the ws. It might not be the best behaviour but it's what we've been using so far in the wallet and wallet lib. We can discuss changing it on the wallet lib refactor.